src/Authentication/Authentication.Core/TokenCache/InMemoryTokenCacheOptions.cs

```diff
@@ -0,0 +1,76 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using Azure.Identity;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Graph.PowerShell.Authentication.Core.TokenCache
+{
+    public class InMemoryTokenCacheOptions : UnsafeTokenCacheOptions
+    {
+        private readonly ReaderWriterLockSlim _sessionLock = new ReaderWriterLockSlim(LockRecursionPolicy.SupportsRecursion);
+        internal ReadOnlyMemory<byte> TokenCache { get; private set; }
```

Comment:
seems odd that the options class is holding the reference to the cache and operating on the object.

Reply:
Good point! It is a cache manager. It's Azure Identity that named the base class [UnsafeTokenCacheOptions](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.unsafetokencacheoptions?view=azure-dotnet). I opted to use the options suffix to be in sync with Azure Identity naming.